ndb/tests/conftest.py

```diff
@@ -40,3 +43,28 @@ def reset_state():
     yield
     model.Property._FIND_METHODS_CACHE.clear()
     model.Model._kind_map.clear()
+
+
+@pytest.fixture
+def environ():
+    """Copy of ``os.environ``"""
+    original = os.environ.copy()
+    environ = original.copy()
+    os.environ = environ
+    yield environ
+    os.environ = original
+
+
+@pytest.fixture(autouse=True)
+def initialize_environment(request, environ):
+    """Set environment variables to default values.
+
+    There are some variables, like ``GOOGLE_APPLICATION_CREDENTIALS``, that we
+    want to reset for unit tests but not system tests. This fixture introspects
+    the current request, determines whether it's in a unit test, or not, and
+    does the right thing.
+    """
+    if request.module.__name__.startswith("tests.unit"):
```

Comment:
TIL the [`request`][1] fixture, thanks!

[1]: https://docs.pytest.org/en/latest/reference.html#request

Reply:
:+1: